setup.py

```diff
@@ -1,122 +1,129 @@
-    long_description=long_description,
-    license='BSD-style license',
-    platforms=['any'],
-    author='Vadim Lebedev, Ben Timby, Travis Cunningham, Ryan Johnston, SmartFile',
-    author_email='tech@smartfile.com',
-    url=repourl,
-    packages=['libarchive'],
-    classifiers=[
-        'Development Status :: 5 - Production/Stable',
-        'Intended Audience :: Developers',
-        'Operating System :: OS Independent',
-        'Programming Language :: C',
-        'Programming Language :: Python',
-        'Topic :: System :: Archiving :: Compression',
-        'Topic :: Software Development :: Libraries :: Python Modules',
-    ],
-    cmdclass={
-        'build_ext': build_ext_extra,
-    },
-    ext_modules=[__libarchive],
-)
+#!/usr/bin/env python
+#
+# Copyright (c) 2015, SmartFile <tcunningham@smartfile.com>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * Neither the name of the organization nor the
+#       names of its contributors may be used to endorse or promote products
+#       derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from os import environ
+
+try:
+    from setuptools import setup, Extension
+    from setuptools.command.build_ext import build_ext
+except ImportError:
+    from distutils.core import setup, Extension
+    from distutils.command.build_ext import build_ext
+
+
+
+# Use a provided libarchive else default to hard-coded path.
+libarchivePrefix = environ.get('LIBARCHIVE_PREFIX')
+if libarchivePrefix is None:
+    import pathlib
+    for i in [ '/usr', '/usr/local', '/opt/local' ]:
+        libdir = pathlib.Path(i) / 'lib'
+        if list(libdir.glob('libarchive.*')):
```

Comment:
This is searching the more historic common paths for libraries. But now in distributions like Ubuntu, there are more paths. Examples are `/usr/local/lib/x86_64-linux-gnu` and  `/usr/lib/x86_64-linux-gnu`. I believe they will always be in subdirectories of the directories above. Changing this to `**/libarchive.*` would handle those cases.

Reply:
ok, made another minor change which is use any instead of list, should be faster as it won't force glob to walk the entire tree when it just needs one valid entry.